### ISSUE TITLE
Search customisations

### ISSUE THREAD

**Issue by mkoppanen** [layerhq/Atlas-iOS#755]
Hello,

first of all: huge fan of Atlas this far. The library has saved me loads of work and works pretty much flawlessly.

My question is a more of a feature request than anything else: I am currently using ATLConversationListViewController to show discussions and the search is initialised by ATLConversationListViewController. What I would like is to have a bit more customisation options for the search: appearance (for some reason when I am tinting the bar it's translucent and my color ends up not matching navigation bar).

Another thing is scope buttons: I would like to toggle the search between different scoped (conversation title / participants and message contents).

**Comment by kcoleman731** [comment]
Hi @mkoppanen 

Thank you very much for your kind works on Atlas. Very nice to hear you are enjoying the library. 

As for your feature request...Can you tell me how you are changing the color of the search bar? I just tested our the following code leveraging UIAppearanceProxy and the color match.
```
[[UISearchBar appearance] setBarTintColor:[UIColor greenColor]];
[[UINavigationBar appearance] setBarTintColor:[UIColor greenColor]];
```

As for scope buttons, could you provide a bit more detail into what you are looking for here? Some designs could help as well.

**Comment by mkoppanen** [comment]
@kcoleman731,

I am using opacity on the navigation bar, I wonder if that's causing my issues. As for the scope buttons I mean the standard approach of segmented control as seen for example in this image: http://cdn4.raywenderlich.com/wp-content/uploads/2014/07/UISearchBar-square.png

It would also be nice to be able to completely disable search (currently I don't see how that would be possible).

**Comment by kcoleman731** [comment]
@mkoppanen 

My apologies for the VERY delayed response. Thanks for your clarification on the scope. That functionality is about out of scope (no pun intended) for the controller. I'd be happy to move the search bar and search display controller properties to the public interface. This would allow to perform your desired customizations in a subclass. Would that work for you?

**Comment by mkoppanen** [comment]
Hey,

and no worries. Sometimes things take time.

Let me just clarify a bit: I was thinking about the scope buttons initially because otherwise it's not necessarily clear what the user is searching for: conversations with specific users? matches of strings in conversations? 

I think your suggestion would take care most of my needs. In addition it would be nice to have a boolean option like "isSearchEnabled" to easily hide the search if it's not needed.

Thanks,
Mikko

**Comment by kcoleman731** [comment]
@mkoppanen Sounds great! I can go ahead and make the change. Another dev also asked for the `searchEnabled` boolean so I can make that happen as well. I'll try and get it on master tomorrow.

**Comment by mkoppanen** [comment]
@kcoleman731,

excellent! Great work guys, really appreciate it.

**Comment by kcoleman731** [comment]
@mkoppanen I started looking at exposing the search bar this AM and realized it is actually unnecessary. 

We set the Search Bar as` self.tableView.tableHeaderView`. As the `ATLConversationListViewController` itself is a `UITableViewController` you already have access to the `tableView` property and are free to override `tableHeaderView`

For your level of customization, this is probably the best approach. Any issues with that?
